router/client/src/sharded_client.rs

```diff
@@ -268,3 +307,68 @@ fn merge_generations(
     }
     Ok((generations, next_batch))
 }
+
+fn format_ner_output(
+    classify_prediction_list: ClassifyPredictionList,
+    tokenizer: Arc<Tokenizer>,
+) -> Vec<Entity> {
+    let input_ids =
+        &classify_prediction_list.input_ids[1..classify_prediction_list.input_ids.len() - 1];
+    let predicted_token_class =
+        &classify_prediction_list.predictions[1..classify_prediction_list.predictions.len() - 1];
+    let scores = &classify_prediction_list.scores[1..classify_prediction_list.scores.len() - 1];
+
+    let tokens: Vec<String> = {
+        let re = Regex::new(r"\b\w+\b|\S").unwrap();
+        re.find_iter(&tokenizer.decode(input_ids, true).unwrap())
+            .map(|m| m.as_str().to_string())
+            .collect()
+    };
+
+    let mut ner_results = Vec::new();
+    let mut current_entity: Option<Entity> = None;
+    for (i, ((token, token_class), score)) in tokens
+        .iter()
+        .zip(predicted_token_class.iter())
```

Comment:
Btw, looks like there's a nice macro for zipping multiple iterators: https://stackoverflow.com/a/29669741

Reply:
oooh good find